lib/r10k/module_loader/puppetfile.rb

```diff
@@ -134,11 +148,60 @@ def add_module(name, module_info)
           return @modules
         end
 
+        # If this module's metadata has a static version and that version
+        # matches the existing module declaration use it, otherwise create
+        # a regular module to sync.
+        unless mod.version && (mod.version == @existing_module_versions_by_name[mod.name])
+          mod = mod.to_implementation
+        end
+
         @modules << mod
       end
 
      private
 
+      def empty_load_output
+        {
+          modules: [],
+          managed_directories: [],
+          desired_contents: [],
+          purge_exclusions: []
+        }
+      end
+
+      def with_readable_puppetfile(puppetfile_path, &block)
+        if File.readable?(puppetfile_path)
+          block.call
+        else
+          logger.debug _("Puppetfile %{path} missing or unreadable") % {path: puppetfile_path.inspect}
+
+          empty_load_output
+        end
+      end
+
+      def manage_inputs(name, info)
```

Comment:
I don't love this method name, it's unclear to me what is being managed, per se. Maybe something like "configure_deploy_options" or something? Or is there another way you were thinking about it?